src/gui/src/PatternEditor/PatternEditorPanel.cpp

```diff
@@ -884,6 +870,47 @@ void PatternEditorPanel::patternSizeChanged( int nSelected )
 	EventQueue::get_instance()->push_event( EVENT_SELECTED_PATTERN_CHANGED, -1 );
 }
 
+void PatternEditorPanel::updatePatternSizeLCD(){
+	// update pattern size
+	int nPatternSize = m_pPattern->get_length();
+	int den = m_pPattern->get_denominator(); //TODO check
+	int nBeatUnit = MAX_NOTES / den;
+
+	char tmp[20];
+	if (nPatternSize % nBeatUnit == 0) {
+		sprintf( tmp, "%d/%d", nPatternSize / nBeatUnit, den);
+	} 
+	else {
+		sprintf( tmp, "%.3f/%d", (float) nPatternSize / nBeatUnit, den);
+	}
+	__pattern_size_LCD->setText( tmp );//
+}
+
+
+void PatternEditorPanel::patternSizeLCDClicked()
+{
+	Hydrogen *pEngine = Hydrogen::get_instance();
+	if ( pEngine->getState() != STATE_READY ) {	
+		QMessageBox::information( this, "Hydrogen", tr( "Is not possible to change the pattern size when playing." ) );
+		return;
+	} //TODO is it really impossible to change the pattern size when playing?
+
+	bool bIsOkPressed;
+	double fNewVal= QInputDialog::getDouble( this, "Hydrogen", tr( "New Size value in quarter-notes" ), m_pPattern->get_length() * 4 / (double) MAX_NOTES, 0, 10000, 5, &bIsOkPressed );
```

Comment:
Also, target line width for formatting is 120 characters. This would read better split over multiple lines, and using named parameters?

Reply:
> With 48 TPQN, one tick is slightly more than 0.02 quarter notes, so specifying 2 decimal places rather than 5 is enough to accurately and unambiguously represent the length. I think it also looks less intimidating to users?

Good analysis, I like the idea even if in some situations I would prefer more than 2 decimal places, for example a 32note is 0.125 quarter notes. I see that input 0.13 makes the same pattern length once rounded (ie 6 ticks), but 48/16 has a finite decimal representation. However this case is highly unlikely

> Also, target line width for formatting is 120 characters. This would read better split over multiple lines, and using named parameters?

Right